.github/workflows/publish_to_maven_local.yml

```diff
@@ -4,12 +4,21 @@ on:
   push:
     branches:
       - main
+    paths:
+      - '**.kt'
+      - '**.kts'
+      - '**.properties'
+      - '**.json'
+      - '**.xml'
+      - '**.yml'
+  pull_request:
   workflow_dispatch:
 
 jobs:
   publish:
     name: Publish pulumi-kotlin to Maven Local Repository
```

Comment:
(minor)

Thinking about this now, encountering such a workflow (I mean, with such a name) might be confusing for newcomers. "Why would you push to Maven local? Agents are cleaned-up anyway." 

Maybe we can express the intent better by improving this name? 

Examples:

- "Test pulumi-kotlin generated code by publishing to Maven Local repository"